src/Fission.re

```diff
@@ -1,87 +1,85 @@
-let baseURL = "http://localhost:1337";
-let env_username = "ca2c70bc13298c5109ee";
-let env_password = "VlBgonAFjZon2wd2VkTR3uc*p-XMd(L_Zf$nFvACpHQShqJ_Hp2Pa";
-
 type cid = string;
-type auth = {
-  username: string,
-  password: string,
+
+let await = promise =>
+  promise
+  |> Js.Promise.then_(response => Js.Promise.resolve(response##data))
+  |> Js.Promise.catch(Js.Promise.resolve);
+
+let octetHeader = Axios.Headers.fromObj({"content-type": "application/octet-stream"});
+
+module type URL = {
+  let baseUrl: string;
+};
+
+module type Auth = {
+  let username: string;
+  let password: string;
 };
 
-let octetHeader =
-  Axios.Headers.fromObj({"content-type": "application/octet-stream"});
+module Simple = (Base: URL) => {
```

Comment:
Executive Summary
* Reason has (roughly) first-class _modules_
* It requires a bit more manual typing, however
* Modules can by typed as well (comment above)
* Each file is implicitly wrapped in a module with the file name
* Modules can contain other modules
* Functions of type `module -> module` are called "functors" in Reason/OCaml
  * Haskell and math have a concept called functor, _which is different_
* Argument must also be a module, since it's happening at the module-level